eng/pipelines/templates/jobs/ci.yml

```diff
@@ -99,36 +110,60 @@ jobs:
           $versioningProperties = "/p:OfficialBuildId=$(OfficialBuildId) /p:SkipDevBuildNumber=$skipDevBuildNumber";
           echo "##vso[task.setvariable variable=VersioningProperties]$versioningProperties"
         displayName: "Setup .NET specific versioning properties"
+
       - task: UsePythonVersion@0
         displayName: 'Use Python 3.11'
         inputs:
           versionSpec: '3.11'
+
       - template: /eng/pipelines/templates/steps/dotnet-diagnostics.yml
         parameters:
           LogFilePath: $(Build.ArtifactStagingDirectory)/pack.binlog
-      - script: >-
-          dotnet pack eng/service.proj -warnaserror
-          /p:ValidateRunApiCompat=true
-          /p:SDKType=${{ parameters.SDKType }}
-          /p:ServiceDirectory=${{ parameters.ServiceDirectory }}
-          /p:IncludeTests=false
-          /p:PublicSign=false $(VersioningProperties)
-          /p:Configuration=$(BuildConfiguration)
-          /p:CommitSHA=$(Build.SourceVersion)
-          /p:ArtifactsPackagesDir=$(Build.ArtifactStagingDirectory)
-          $(DiagnosticArguments)
-        displayName: "Build and Package"
 
-      - task: Powershell@2
-        inputs:
-          filePath: $(Build.SourcesDirectory)/eng/common/scripts/Save-Package-Properties.ps1
-          arguments: >
-            -ServiceDirectory ${{parameters.ServiceDirectory}}
-            -OutDirectory $(Build.ArtifactStagingDirectory)/PackageInfo
-            -AddDevVersion:$$(SetDevVersion)
-          pwsh: true
-          workingDirectory: $(Pipeline.Workspace)
-        displayName: Update package properties with dev version
+      # only use the project list override file if the service directory is not auto
+      - ${{ if eq(parameters.ServiceDirectory, 'auto') }}:
+        - pwsh: |
+            if (!"$(ProjectNames)") {
+              Write-Host "No packages to build."
+              exit 0
+            }
+            else {
+              dotnet pack eng/service.proj -warnaserror `
+                /p:ValidateRunApiCompat=true `
+                /p:SDKType=${{ parameters.SDKType }} `
+                /p:IncludeTests=false `
+                /p:PublicSign=false $(VersioningProperties) `
+                /p:Configuration=$(BuildConfiguration) `
+                /p:CommitSHA=$(Build.SourceVersion) `
+                /p:ArtifactsPackagesDir=$(Build.ArtifactStagingDirectory) `
+                /p:ServiceDirectory=* `
+                /p:ProjectListOverrideFile=$(ProjectListOverrideFile) `
+                $(DiagnosticArguments)
+            }
```

Comment:
```suggestion
            if (!"$(ProjectNames)") {
              Write-Host "No packages to build."
              exit 0
            }
            dotnet pack eng/service.proj -warnaserror `
              /p:ValidateRunApiCompat=true `
              /p:SDKType=${{ parameters.SDKType }} `
              /p:IncludeTests=false `
              /p:PublicSign=false $(VersioningProperties) `
              /p:Configuration=$(BuildConfiguration) `
              /p:CommitSHA=$(Build.SourceVersion) `
              /p:ArtifactsPackagesDir=$(Build.ArtifactStagingDirectory) `
              /p:ServiceDirectory=* `
              /p:ProjectListOverrideFile=$(ProjectListOverrideFile) `
              $(DiagnosticArguments)
```

Reply:
I sometimes just don't even know where my brain is.